src/middleware/graphql/graphqlish.service.js

```diff
@@ -0,0 +1,161 @@
+              (coding) => claimTypesToBeMatched.indexOf(coding.code) !== -1
+            );
+          }
+          if (!claimTypeMatches) {
+            return false;
+          }
+          let billablePeriodMatches = true;
+          if (billablePeriod) {
+            if (
+              billablePeriod.start &&
+              ((_a = bundleEntry.resource.billablePeriod) === null || _a === void 0
+                ? void 0
+                : _a.start)
+            ) {
+              const startDate = new Date(bundleEntry.resource.billablePeriod.start);
+              startDate.setHours(0, 0, 0, 0);
+              const filterStartDate = new Date(billablePeriod.start);
+              filterStartDate.setHours(0, 0, 0, 0);
+              billablePeriodMatches = startDate >= filterStartDate;
+            }
+            if (
+              billablePeriodMatches &&
+              billablePeriod.end &&
+              ((_b = bundleEntry.resource.billablePeriod) === null || _b === void 0
+                ? void 0
+                : _b.end)
+            ) {
+              const endDate = new Date(bundleEntry.resource.billablePeriod.end);
+              endDate.setHours(0, 0, 0, 0);
+              const filterEndDate = new Date(billablePeriod.end);
+              filterEndDate.setHours(0, 0, 0, 0);
+              billablePeriodMatches = endDate <= filterEndDate;
+            }
+          }
+          if (!billablePeriodMatches) {
+            return false;
+          }
+          let allowedAmountMatches = true;
+          if (allowedAmount) {
+            if (!isNaN(allowedAmount.min)) {
+              if (allowedAmount.min <= 0) {
+                allowedAmountMatches =
+                  bundleEntry.resource.amounts.allowed >= allowedAmount.min ||
+                  bundleEntry.resource.amounts.allowed === null;
+              } else {
+                allowedAmountMatches = bundleEntry.resource.amounts.allowed >= allowedAmount.min;
+              }
+            }
+            if (allowedAmountMatches && !isNaN(allowedAmount.max)) {
+              if (allowedAmount.max >= 0) {
+                allowedAmountMatches =
+                  bundleEntry.resource.amounts.allowed <= allowedAmount.min ||
+                  bundleEntry.resource.amounts.allowed === null;
+              } else {
+                allowedAmountMatches = bundleEntry.resource.amounts.allowed <= allowedAmount.min;
+              }
+            }
+          }
+          if (!allowedAmountMatches) {
+            return false;
+          }
+          let youPaidMatches = true;
+          if (youPaid) {
+            if (!isNaN(youPaid.min)) {
+              youPaidMatches = bundleEntry.resource.amounts.memberPaid >= youPaid.min;
+            }
+            if (youPaidMatches && !isNaN(youPaid.max)) {
+              youPaidMatches = bundleEntry.resource.amounts.memberPaid <= youPaid.max;
+            }
+          }
+          if (!youPaidMatches) {
```

Comment:
Since this logic is used in L85, L61, L30, could we move it to a single IF check or a separate function like a pre-validation call to help with unit tests in the future?